crates/rpc-types/src/transaction/request.rs

```diff
@@ -90,6 +90,22 @@ impl OpTransactionRequest {
         self
     }
 
+    /// Return the tx type this request can be built as. Computed by checking
+    /// the preferred type, and then checking for completeness.
+    pub fn buildable_type(&self) -> Option<OpTxType> {
+        if self.0.chain_id.is_none() {
+            return Some(OpTxType::Deposit);
+        }
+        match self.0.buildable_type() {
+            Some(TxType::Legacy) => Some(OpTxType::Legacy),
+            Some(TxType::Eip1559) => Some(OpTxType::Eip1559),
+            Some(TxType::Eip2930) => Some(OpTxType::Eip2930),
+            Some(TxType::Eip7702) => Some(OpTxType::Eip7702),
+            Some(TxType::Eip4844) => Some(OpTxType::Eip1559),
```

Comment:
how did you think here, eip4844 -> eip1559 ? I would have thought that variant is just skipped, possibly with a debug log message that eip4844 isn't supported for eip4844

Reply:
We can do that - was pretty unsure